dvc/repo/experiments/remove.py

```diff
@@ -33,6 +49,33 @@ def remove(repo, exp_names=None, queue=False, **kwargs):
     return removed
 
 
+def _clear_stash(repo):
+    removed = len(repo.experiments.stash)
+    repo.experiments.stash.clear()
+    return removed
+
+
+def _clear_workspace(repo):
+    head = repo.scm.get_ref("HEAD")
+    ref_infos = list(exp_refs_by_baseline(repo.scm, head))
+    remove_exp_refs(repo.scm, ref_infos)
+    stashed = 0
+    stash_revs = repo.experiments.stash_revs
+    for _, ref_info in stash_revs.items():
+        if head == ref_info.baseline_rev:
+            repo.experiments.stash.drop(ref_info.index)
+            stashed += 1
+    return stashed + len(ref_infos)
+
+
+def _clear_all_baseline(repo):
+    ref_infos = list(exp_refs(repo.scm))
+    remove_exp_refs(repo.scm, ref_infos)
+    removed = len(repo.experiments.stash)
+    repo.experiments.stash.clear()
```

Comment:
I'm not actually sure if `-A/--all` should affect queued experiments, since we have a separate option for that (`--queue`). It seems to mie that `-A` should only affect "full" experiments that have already been reproduced

maybe a question for @dberenbaum

Reply:
🤔 It seems simpler to explain if it includes the queue, but it's probably more useful if it doesn't since we already have an option to remove the whole queue. Going back to the discussion above, `dvc exp list` ignores queued experiments, and I think it makes sense to do the same here.

Reply:
@dberenbaum , Same with `-w/--workspace`, do not affect queued ones? 

table | workspace based | all based
------ | ------ | -------------
committed | 1 | 2
queued | 3  | 4

currently `--queue` removes `3` and `4`. After next change `--all` removes `1` and `2`, and workspace? removes `1` and `3` or only `1`

Reply:
Sorry, I thought we were getting rid of `--workspace`?

Reply:
I keep both of them for now.

Reply:
I would recommend keeping one, and I prefer `--all` so we don't have to get into the confusing corner cases like the one you mentioned above. I think when you document the features, you will find it easier if you don't have to document `--workspace` 😄 .

As mentioned in the other discussion above, I think you can add `--rev` later if functionality similar to `--workspace` is needed.